tests/test_osut.py

```diff
@@ -5010,6 +5010,180 @@ def test34_generated_skylight_wells(self):
+        a_roof_north   = list(a_roof_north)
+        c_core_ceiling = list(c_core_ceiling)
+        a_roof_north.reverse()
+        c_core_ceiling.reverse()
+        self.assertFalse(openstudio.join(a_roof_north, c_core_ceiling, TOL2))
+        self.assertFalse(openstudio.intersect(a_roof_north, c_core_ceiling, TOL))
+
+        # A future revision of OSut's 'roofs' and 'grossRoofArea' would require:
+        # - a new method identifying leader lines amongst surface vertices
+        # - a new method identifying surface cutouts amongst surface vertices
+        # - a method to prune both leader lines and cutouts from surface vertices
+        # - have 'roofs' & 'grossRoofArea' rely on the remaining outer vertices
+        #   ... @todo?
+        self.assertEqual(o.status(), 0)
+        del model
+
+        # --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- --- #
+        # CASE 2:
+        path  = openstudio.path("./tests/files/osms/in/smalloffice.osm")
+        model = translator.loadModel(path)
+        self.assertTrue(model)
+        model = model.get()
+
+        core  = model.getSpaceByName("Core_ZN")
+        attic = model.getSpaceByName("Attic")
+        self.assertTrue(core)
+        self.assertTrue(attic)
+        core  = core.get()
+        attic = attic.get()
+
+        # Tag attic as an INDIRECTLY-CONDITIONED space.
+        key = "indirectlyconditioned"
+        val = core.nameString()
+        self.assertTrue(attic.additionalProperties().setFeature(key, val))
+        self.assertFalse(osut.arePlenums(attic))
+        self.assertFalse(osut.isUnconditioned(attic))
+        self.assertAlmostEqual(osut.setpoints(attic)["heating"], 21.11, places=2)
+        self.assertAlmostEqual(osut.setpoints(attic)["cooling"], 23.89, places=2)
+
+        # Here, GRA includes ALL plenum roof surfaces (not just vertically-cast
+        # roof areas onto the core ceiling). More roof surfaces == greater
+        # skylight areas to meet the SRR% of 5%.
+        gra_plenum = osut.grossRoofArea(core)
+        self.assertAlmostEqual(gra_plenum, total1, places=2)
+
+        rm2 = osut.addSkyLights(core, dict(srr=srr))
+        if o.logs(): print(o.logs())
+        self.assertAlmostEqual(rm2, total1, places=2)
+
+        # The total skylight area is greater than in CASE 1. Nonetheless, the
+        # method is able to meet the requested SRR 5%. This may not be
+        # achievable in other circumstances, given the constrained roof/core
+        # overlap. Although a plenum vastly larger than the room(s) it serves is
+        # rare, it remains certainly problematic for the application of the
+        # Canadian NECB reference building skylight requirements.
+        core_skies = osut.facets(core, "Outdoors", "Skylight")
+        sky_area2  = sum([sk.grossArea() for sk in core_skies])
+        self.assertAlmostEqual(sky_area2, 29.94, places=2)
+        ratio2     = sky_area2 / rm2
+        self.assertAlmostEqual(ratio2, srr, places=2)
```

Comment:
Converting _SmallOffice_ UNCONDITIONED "attic" to an INDIRECTLY-CONDITIONED "plenum" (or similar dead-air space). Greater roof surface areas? Larger skylights + wells (for the same 5% SRR):

<img width="985" height="678" alt="offplenum_sky" src="https://github.com/user-attachments/assets/d08cf3f7-0216-4aa5-b2c2-34b974f25deb" />

Tests are green. Same (single) _eplusout.err_ warning on coincident/collinear vertices - that's fine.